src/azure-cli/azure/cli/command_modules/acr/_stream_utils.py

```diff
@@ -92,10 +93,18 @@ def _stream_logs(no_format,  # pylint: disable=too-many-locals, too-many-stateme
     # Try to get the initial properties so there's no waiting.
     # If the storage call fails, we'll just sleep and try again after.
     try:
-        props = blob_service.get_blob_properties(
+        # Need to call "exists" API to prevent storage SDK logging BlobNotFound error
+        log_exist = blob_service.exists(
             container_name=container_name, blob_name=blob_name)
-        metadata = props.metadata
-        available = props.properties.content_length
+
+        if log_exist:
+            props = blob_service.get_blob_properties(
+                container_name=container_name, blob_name=blob_name)
+            metadata = props.metadata
```

Comment:
@northtyphoon can `props` be null?

Reply:
no, it will throw error like AureHttpError, in that case, we will retry or just exit if the code doesn't know how to handle it